### DISCUSSION TITLE
Pocketbase Go Framework: Do I need locks?

### DISCUSSION THREAD

**Discussion by ramadhanafif** [pocketbase/pocketbase#4970] [Q&A] [answered]
I have several question in regards to using pocketbase as Golang framework.

1. Does pocketbase need locks when accessed by multiple goroutine? Can I safely run a custom query using `Dao().DB().NewQuery()` ?
2. Does SQLite need locks when accessed by multiple goroutine? If I remembered correctly, pocketbas is using WAL mode on SQLITE, is this true?

**Comment by ganigeorgiev** [comment, marked as answer]
Users are not expected to perform explicit db locks.
SQLite in WAL mode allows multiple readers and a single writer (other writes queue up).
I don't exactly understand your concerns and what actually is the issue you are experiencing but you can safely execute concurrently raw queries using `app.Dao().DB()`.

**Reply by ramadhanafif** [reply]
That answers it. Thanks!
